runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -451,11 +451,14 @@ function LanguageTree:parse(range)
   return self._trees
 end
 
+---@deprecated Misleading name. Use `LanguageTree:children()` (non-recursive) instead,
+---            add recursion yourself if needed.
 --- Invokes the callback for each |LanguageTree| and its children recursively
 ---
 ---@param fn fun(tree: LanguageTree, lang: string)
 ---@param include_self boolean|nil Whether to include the invoking tree in the results
 function LanguageTree:for_each_child(fn, include_self)
+  vim.deprecate('LanguageTree:for_each_child()', 'LanguageTree:children()', '0.11')
```

Comment:
The [deprecation policy we've decided upon](https://github.com/neovim/neovim/blob/master/MAINTAIN.md#deprecating-and-removing-features) dictates first adding only a `@deprecated` annotation and waiting until the next release to add `vim.deprecate`, and finally removing the release after.

So this `vim.deprecate` should be removed, and in the 0.11 release cycle this should be added indicating that `for_each_child` will be removed in 0.12.

Reply:
We need to add flexibility to that deprecation policy. We should allow judgment for cases where we want to be more aggressive.

Especially for treesitter support which is *explicitly* declared *experimental*. We can't say things are experimental and then insist that we go through full deprecation cycles for them.

Reply:
I don't disagree, but I think our deprecation policy should state that explicitly. The point of having it is that our process (whatever that is) is documented clearly, if our process and the documentation don't align then that's a problem.

Reply:
https://github.com/neovim/neovim/pull/25159

Reply:
100% agree and thanks for the callout.